File: Polynomial Regression/polynomial_regression.py
# -*- coding: utf-8 -*-
"""
Created on Sun Apr 14 22:21:05 2019

@author: saranya.ravichandran
"""

#import libraries
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

#import data
df = pd.read_csv('Position_Salaries.csv')
X = df.iloc[:,1:2].values
y = df.iloc[:,2].values

#treat missing data --> not needed here
#treat categorical data --> not needed here
#split into train and test data --> not needed here as dataset has only 10 values.

#linear regression
from sklearn.linear_model import LinearRegression
linear_reg = LinearRegression()
linear_reg.fit(X,y)

#polynomial regression
from sklearn.preprocessing import PolynomialFeatures
poly_reg = PolynomialFeatures(degree = 4)
X_poly = poly_reg.fit_transform(X)
poly_reg.fit(X_poly,y)
lin_reg2 = LinearRegression()
lin_reg2.fit(X_poly,y)

#plotting for linear regression
plt.scatter(X,y,color = 'red')
plt.plot(X,linear_reg.predict(X),color = 'blue')
plt.title('Truth or bluff (Linear regression)')
plt.xlabel('Position')
plt.ylabel('Salary')
plt.show()

#plotting for polynomial regression
X_grid = np.arange(min(X),max(X),0.1)
X_grid = X_grid.reshape((len(X_grid),1))
plt.scatter(X,y,color='red')
plt.plot(X_grid,lin_reg2.predict(poly_reg.fit_transform(X_grid)),color='blue')
plt.title('Truth or bluff (polynomial regression)')
plt.xlabel('Position')
plt.ylabel('Salary')
plt.show()

File: Polynomial Regression/polynomial_regression.R
# import data

df = read.csv('Position_Salaries.csv')
#index starts at 1 for R
df = df[2:3]

#treat missing data -->not needed here
#treat categorical data --> not needed here
#split into train and test data --> not needed here as dataset is small
#feature scaling --> not needed here

#linear regression
lin_reg = lm(formula = Salary ~ ., data = df)
#install.packages("ggplot2")
#plot data
library(ggplot2)
ggplot() +
  geom_point(aes(x = df$Level, y = df$Salary),color = 'red') +
  geom_line(aes(x = df$Level, y = predict(lin_reg,newdata = df)),color = 'blue') +
  ggtitle('Linear regression(Level Vs Salary') +
  xlab('Level') +
  ylab('Salary')

#polynomial regression
df$Level1 = df$Level^2
df$Level2 = df$Level^3
df$Level3 = df$Level^4

poly_reg = lm(formula = Salary ~ ., data = df)
#install.packages("ggplot2")
#plot data
library(ggplot2)
ggplot() +
  geom_point(aes(x = df$Level, y = df$Salary),color = 'red') +
  geom_line(aes(x = df$Level, y = predict(poly_reg,newdata = df)),color = 'blue') +
  ggtitle('Polynomial regression(Level Vs Salary)') +
  xlab('Level') +
  ylab('Salary')

#predicting new results with polynomial regression
y_predict = predict(poly_reg,data.frame(Level = 6.5, Level1 = 6.5^2,
                                      Level2 = 6.5^3, Level3 = 6.5^4))